internal/lifecycle/lifecycle.go

```diff
@@ -40,22 +41,39 @@ func (l Lifecycle) Execute(commandsDir string) error {
 	return nil
 }
 
-// NewLifecycle returns a lifecycle built with reasonable defaults based on whether the sample is java-based
+// NewLifecycle tries to parse the different options provided for build and deploy command configuration. If none of
+// those options are set up, it falls back to reasonable defaults based on whether the sample is java-based
 // (has a pom.xml) that doesn't have a Dockerfile or isn't.
 func NewLifecycle(sampleDir, serviceName, gcrURL string) Lifecycle {
+	readmePath := fmt.Sprintf("%s/README.md", sampleDir)
+
+	_, err := os.Stat(readmePath)
+	readmeExists := err == nil
+
+	if readmeExists {
```

Comment:
nit: I think you can remove this variable from scope altogether now 
```suggestion
	

	if _, err := os.Stat(readmePath); err != nil {
```

Reply:
Resolved, thank you.